test/CMakeLists.txt

```diff
@@ -37,8 +37,11 @@ endif ()
 find_package(GTest)
 if (${GTest_FOUND})
     # workaround for github.com/conan-io/conan-center-index/issues/3222
-    set(CNL_GTEST_MAIN_TARGET GTest::Main CACHE STRING "alternative GTest target name")
-
+    if(${CMAKE_SYSTEM_NAME} MATCHES "Darwin")
```

Comment:
I figure that if no OSes or compilers are listed in the CMakeLists.txt, then the build system is more broadly-accepting of tool chains, more portable, and more robust to change over time.

In the case of this particular hack, I decide [from outside the build system](https://github.com/johnmcfarlane/cnl/blob/26759304fb0bf03b464df7060e039eba6c461031/conanfile.py#L73) (i.e. outside the CMakeLists.txt files) whether I need to fudge this particular target name variable. The pattern seems to be: if it's Conan, use _GTest::gtest_main_, if it's not Conan, use _GTest::Main_. 

Note: you may be witnessing a different conditions because you are using a different Conan generator.

Reply:
BTW, here's [the issue](https://github.com/conan-io/conan/issues/7039#issuecomment-739585712) for which `CNL_GTEST_MAIN_TARGET` is the workaround.